### ISSUE TITLE
Codecgen script generates incorrect codec

### ISSUE THREAD

**Issue by pmorie** [kubernetes/kubernetes#15892]
@wojtek-t In #15352, I found that hack/update-codecgen.sh can generate incorrect codecs.  If you pull my `fsgroup` branch and run `hack/test-go.sh pkg/api` you will find that all ObjectMeta instances in the extensions api drop their name fields on decoding.

**Comment by pmorie** [comment]
@wojtek-t I also ran into the same issue when working on #14705.  You can grab my `pod-sc-inline` branch to recreate with that one.  I hope I am doing something wrong, but if I am not, I'm basically blocked on this and need to get it working asap.

**Comment by wojtek-t** [comment]
@pmorie I'm not sure what you did (it's possible that there is something wrong with the scripts, but I cannot reproduce it) - but the problem is with your PR.

So to clarify:
- in #15352 you completely removed the following files:
 a) pkg/api/v1/types.generated.go
 b) pkg/apis/extensions/types.generated.go
 c) pkg/apiserver/testing/types.generated.go
- in #14705 you completely removed the following files:
 a) pkg/api/types.generated.go
 b) pkg/apis/extensions/v1beta1/types.generated.go

This is the reason why it doesn't work.

Note that currently the hack/updage-codecgen.sh script is just updating the files that already exist - so running it once again will not help - you need to create those manually (possibly with some rubbish) and regenerate them.

**Comment by pmorie** [comment]
@wojtek-t I think you're right.  I might file a follow-up issue to smooth out some rough edges (or open a PR), but I think I was doing it wrong.  Thanks a lot for taking a look!

**Comment by wojtek-t** [comment]
@pmorie - thanks! please don't hesitate to open an issue if you observe some problems with the scripts.
